### ISSUE TITLE
Link open behaviour

### ISSUE THREAD

**Issue by Ninerian** [trigger-corp/trigger.io-tabs#29]
Hello Antoine,

currently links, opened in a tabs module, are loaded in the same tab and replacing the website. Even if the link has a `target="_blank"` set. 

Is there the possibility to modify this behaviour, so that these links are opened outside the app?

**Comment by antoinevg** [comment]
We've had this conversation before 😄

**tl;dr:** `WKWebView` doesn't respond to target="_blank" links - this is a Safari Web Browser behaviour.

That said, if you wanted to implement something like this yourself then you'll want to look at the `tabs_WKWebViewDelegate::decidePolicyForNavigationAction` method.

This is where the tabs module decides how to respond to navigation actions.

So off the top of my head I can think of two ways you could implement this functionality:

**Easier:** Extend the tabs module to have a configurable list of URL patterns that will be opened outside the app. You can then match against these patterns in `decidePolicyForNavigationAction` and pass matching URL's to `[[UIApplication sharedApplication] openURL:url];`

**Harder:** Execute some Javascript inside the tabs when they load that will iterate through all `<a href>` tags on the page, check the `target` attribute and then call into the tabs module to give it a list of links to be handled specially in `decidePolicyForNavigationAction`.

**Comment by antoinevg** [comment]
Oooh, looks like there might also be some new options with `WKWebView`:

You may want to try this first:

https://stackoverflow.com/questions/25713069/why-is-wkwebview-not-opening-links-with-target-blank
